obspy/core/inventory/response.py

```diff
@@ -767,68 +767,84 @@ def __init__(self, resource_id=None, instrument_sensitivity=None,
             msg = "response_stages must be an iterable."
             raise ValueError(msg)
 
-    def recalculate_overall_sensitivity(self):
+    def recalculate_overall_sensitivity(self, frequency=None):
         """
         Recalculates the overall sensitivity.
+
+        :param frequency: Choose frequency at which to calculate the
+            sensitivity. If not given it will be chosen automatically.
         """
         if not hasattr(self, "instrument_sensitivity"):
             msg = "Could not find an instrument sensitivity - will not " \
                   "recalculate the overall sensitivity."
             warnings.warn(msg)
-        elif not self.instrument_sensitivity.input_units:
+            return
```

Comment:
Could be worth a thought to turn this into an Exception instead of warning message and nothing done?